### ISSUE TITLE
getting unexpected result from customize pre-Training model form Keras (Xception model) in C++ after freezing the model .pb .

### ISSUE THREAD

**Issue by kerolos** [tensorflow/tensorflow#30875]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution  (debian 9.1):
- TensorFlow installed from (source):
- TensorFlow version (13.1):
- Python version: python 3.5
- Bazel version (11):
- CUDA/cuDNN version: 10
- GPU model and memory: 32 GB, GTX 1080 TI

**Describe the current behavior**
I am wondering why I can not get the same result when i run the frozen model in c++. I do not know exactly if the tf.keras does preprocessing step when I load the model.  

**Python code: **
-  creat tfrecord:
def get_example_object(image_string, label): ##label is in integer value( from 1 to number of classes )
  feature = {'image':  _bytes_feature(tf.compat.as_bytes(image_string.tostring())),
           'label':  _int64_feature(int(label))}
  return tf.train.Example(features=tf.train.Features(feature=feature))
image = cv2.imread(img_path)
image= cv2.resize(image, (required_height, required_width))

-  training:
def XceptionCNN(IMG_SIZE,OUTPUT_SIZE):

    INPUT_SHAPE = (IMG_SIZE, IMG_SIZE,3)
    input = tf.keras.layers.Input(
        shape=INPUT_SHAPE, dtype=tf.float32,name='input') # 

    Xception_model = Xception(input_tensor=input, include_top=False,weights=None) 
    print('XceptionCNN Model loaded.')
    print(Xception_model.summary())
    x = Xception_model.output    
    x = GlobalAveragePooling2D()(x)
    return model
def _parse_function(proto):
    image_size=150
    class_number =16
    keys_to_features = {'image': tf.FixedLenFeature([], tf.string),
                        "label": tf.FixedLenFeature([], tf.int64)}

    parsed_features = tf.parse_single_example(proto, keys_to_features)
    image = tf.decode_raw(parsed_features['image'], tf.uint8)
    image = tf.cast(image, tf.float32) * (1. / 255)
    image = tf.reshape(image, shape=(image_size, image_size, 3))
 
    label = tf.cast(parsed_features["label"], tf.int32)
    onehot_labels = tf.one_hot(tf.cast(label, tf.int32), class_number)
    return image, onehot_labels

def get_dataset(filepath,epochs, batch_size ):
    dataset = tf.data.TFRecordDataset(filepath)
    dataset = dataset.map(_parse_function, mt.cpu_count())
    dataset = dataset.repeat(epochs).shuffle(1000).batch(batch_size)
    return dataset

**CPP code: **
-  creat float tensor:
        cv::cvtColor(rgbImage, inputImage, CV_RGB2BGR);
        cv::Mat img(imageTargetSize, CV_32FC1, dataTensor);
         resizedPic.convertTo(img, CV_32FC3, 1.0 / 255, 0);
        dataTensor += imageTargetSize.width * imageTargetSize.height * depth;

**output**
- The Training process:
2019-07-18 13:07:35.544638: I tensorflow/stream_executor/dso_loader.cc:152] successfully opened CUDA library libcublas.so.10.0 locally
2306/2307 [============================>.] - ETA: 0s - loss: 0.0218 - acc: 0.9935     
Epoch 00001: val_loss improved from inf to 0.00693, saving model to /mnt/4TB_HD/Training/flaps_keras/txt/model/class_net_normal/weights
2307/2307 [==============================] - 1074s 466ms/step - loss: 0.0217 - acc: 0.9935 - val_loss: 0.0069 - val_acc: 0.9979
Epoch 2/4
2306/2307 [============================>.] - ETA: 0s - loss: 0.0074 - acc: 0.9981       
Epoch 00002: val_loss did not improve from 0.00693
2307/2307 [==============================] - 1029s 446ms/step - loss: 0.0074 - acc: 0.9981 - val_loss: 0.0162 - val_acc: 0.9968
Epoch 3/4
2306/2307 [============================>.] - ETA: 0s - loss: 0.0053 - acc: 0.9988   
Epoch 00003: val_loss improved from 0.00693 to 0.00427, saving model to /mnt/4TB_HD/Training/flaps_keras/txt/model/class_net_normal/weights
2307/2307 [==============================] - 1008s 437ms/step - loss: 0.0053 - acc: 0.9988 - val_loss: 0.0043 - val_acc: 0.9991
Epoch 4/4
2306/2307 [============================>.] - ETA: 0s - loss: 0.0046 - acc: 0.9991       
Epoch 00004: val_loss did not improve from 0.00427
2307/2307 [==============================] - 1012s 439ms/step - loss: 0.0046 - acc: 0.9991 - val_loss: 0.0090 - val_acc: 0.9986
tf_input: Tensor("input:0", shape=(?, 150, 150, 3), dtype=float32)
model_input: <tensorflow.python.keras.engine.training.Model object at 0x7f8fac3f2f98>
tf_output: Tensor("dense/Softmax:0", shape=(?, 16), dtype=float32)
model_output: dense/Softmax

**Comment by ravikyram** [comment]
@kerolos 
Looks like the code is incomplete.Can you please provide full code snippet to reproduce it on our environment.Thanks!

**Comment by ravikyram** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
